### PR TITLE
Draft for hiring engs

### PR DESCRIPTION
**Pull request by dschenkelman**
# MERGE ON TUESDAY, MARCH 22ND
Would like some help to get this merged. 

* @rickyrauch @vctrfrnndz who (either you or someone else) can help me with a better looking version of this?

* Wouldn't change the contents for SEO, the goal is not to SEO, but to get more and better CVs.
* @woloski @mszklanny review
* who can upload the images to the CDN?

### CODE REVIEW

**Comment by vctrfrnndz** [comment]
@dschenkelman Barbara can help you out with this and / or other blog assets.

**Comment by dschenkelman** [comment]
@barbiemmc excellent thanks @vctrfrnndz

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+- Hiring
+---
+
+# tl;dr
+![hiring process](https://cldup.com/WAb9JDiJ3c.png)
+
+# The full version
+Hiring is a key aspect of helping Auth0 grow. We are doing great, there's lots of things we want to get done in the short term and building a team that can handle this growth is paramount for continuing success.
+
+There are many of blog posts explaining why [startups shouldn't handle hiring like big companies do](https://zachholman.com/posts/startup-interviewing-is-fucked/). And we totally agree with this: we want to invest time to take the decision of making a new person part of our team.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: There are many blog posts...

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+## In the beginning
+Usually conversations start when we get an email at [jobs+engineer@auth0.com](mailto:jobs+engineer@auth0.com). This is a very early stage and we have no information other than a CV and the email's contents.
+The first thing we look for is:
+>Does it seem like person WANTS to work at Auth0?
+
+This is a very important point as we believe it is important to have teammates that are passionate about the job they are doing. One of the things we do at this stage is check:
+>Do the email contents reflect interest in the company and position or is this just a template that gets sent to a bunch of potential employers?
+
+After that we take a look at the CV to _try_ to assess relevant experience and think if the candidate could do the job. If we think either **Strong yes** or are in doubt we move on to the next step.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: Whether it is a **strong yes** or we are in doubt, we will move to the next step.

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+## The screener
+The next thing do is learn more about the candidate. To do this we send them a short screener an idea we got from [this intercom post](https://blog.intercom.io/how-we-hire-engineers-part-1/) with questions that look like this (they might vary a bit depending on the specific position):
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: To do this, we send them a short screener--an idea we got from [this intercom post](https://blog.intercom.io/how-we-hire-engineers-part-1/)--with questions that look like this (they might vary a bit depending on the specific position)

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+We also ask them to put together a small [webtask](https://webtask.io/) hack, something that should take a couple of hours.
+
+At this point very few people have thought that investing time to answer this email is not worth it and let us know. More often, we tend to get really interesting responses and learn many things such as:
+
+* The kind of topics this person is interested in.
+* The things they know and like about Auth0: we are a company with a lot of available content, so there's lots of room to share knowledge here.
+* An initial impression at their coding skills.
+
+> For candidates about which we were in doubt about, this is a chance to impress us.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: For candidates that we are still in doubt, this is the chance to impress us.

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+## The interviews
+Assuming that everything went well, I have a call with candidates. The goal is to:
+
+* Go through past work experience
+* Explain the position to the candidate and gauge their interest
+* Understand what they like to do and what [team](https://auth0.com/jobs/engineer) would be a good fit
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: and which team will be a good fit for them.

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+* Evaluate the technical experience, more on "how?" below
+
+Up next we do another call with the engineering lead for the position (e.g. frontend, backend, native, etc.) to both validate technical experience and see if they would like to work with the candidate.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: Up next, we do another call with the engineering lead for the position (e.g. frontend, backend, native, etc.). This is to validate the technical experience and to see if they would like to work with the candidate.

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+### No live coding
+We believe it is really hard to assess technical knowledge with a short live coding exercise. We have all been there before: [it is VERY easy to get nervous](http://www.hanselman.com/blog/HowDoYouDealWithAnxietyWhenLiveCodingInTechnicalInterviews.aspx), it's very hard to define a problem that is actually related to the job and solvable in a short amount of time, etc.
+
+What we do is talk. We go over situations that we have faced at Auth0 and have discussions about their implementation. It doesn't require any specific tool knowlege, it is all very conceptual and the idea is to give candidates room to ask questions about what they don't understand and go over pros and cons of each possible solution.
+
+We believe this has worked out well so far. Being in an interview is enough to make people nervous even without requiring live coding. Most candidates have expressed a sign of relief when being explained that they are not going to have to do that. We actually try to state that at the beginning of the interview as it seems to relax them.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: We believe that this has worked well so far. ..... Most candidates have expressed relief when being explained that they are not going to have to do that. We actually try to state that at the beginning of the interview, as it seems to relax them.

Comment (kukicadnan) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: I loved the approach of having of the interviewee do an actual task that they'll be doing if hired as an exercise. Was able to do it on my own time and interact with the team as if I was already working there so got a great feel for the culture right off the bat.

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+We know this way of evaluating knowledge is not perfect and are continuosly improving based on feedback, but we need a way to try to determine if this person is someone we want to invest more time on for the technical exercise.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: We know that this way....  if this person is someone we want to invest time for the technical exercise.

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+## The technical exercise
+After the two interviews, we move on to the technical exercise. This consists of:
+
+1. Setting up a slack channel with everyone involved in the hiring process.
+2. Providing the exercise (which varies depending on the specific position)
+3. Explain to candidates that they are free to ask questions and validate their thoughts in the channel
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: Explaining candidates ...

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+The exercise takes between 8 and 12 hours to complete and we give ~7 days to do so. Each person approaches this differently and is in charge of managing their own schedule.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: between 8 to 12 hours to complete, and we give

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+What we do is guide them, by going deeper into the requirements or highlighting problems that might lie ahead. We do our best to adjust to each schedule, such as replying on late nights and weekends as we know this requires time from both parties. It is a pretty interactive process!
+
+## The demo and chat with our CTO
+After the exercise is finished we setup a 1 hour call.
+
+During the first half, a 30 minute demo is held to go over the solution's implementation. This is a good point to go over any thing that wasn't previously covered in chat.
+
+If everything goes well, the second half of the our is spent on a chat with Matias, our CTO. He is very involved in the hiring process (actually part of the channels) and wants to get to talk to all potential team members. This is a great opportunity to ask him questions like: "How do you keep your hair so shiny?"
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: the remaining 30 minutes are spent...

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+## Some stats
+To give you an idea of how this all fits together, the following chart shows a summary of candidates that advanced to each stage for a particular job posting:
+![stats](https://cldup.com/uV9cHSZsMl.png)
+
+## Wrapping up
+
+This is what works for us now; we know there's room for improvement. We continuosly adjust the technical discussions and exercises based on feedback from candidates. In fact, one of the reasons for publishing this post is that some candidates wanted to know about our hiring process before they did.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: This is what currently works for us; and we know that there is room for improvement. We continuously adjust..
In fact, one of the reasons for publishing this post is that some candidates wanted to know about our hiring process before they went through it.

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+Hopefully this adds more transparency to our recruiting process.
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: Reworked conclusion: :grin: 
At Auth0 we value transparency, and to practice what we preach we want to be transparent about our recruiting process. So if at any time you want to be a part of Auth0's engineering team, you now have a grasp of what lies ahead. :D

File: _posts/2016-03-15-how-we-hire-engineers.markdown [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+## The screener
+The next thing do is learn more about the candidate. To do this we send them a short screener an idea we got from [this intercom post](https://blog.intercom.io/how-we-hire-engineers-part-1/) with questions that look like this (they might vary a bit depending on the specific position):
+
+1. What's your proudest professional achievement? It can be a personal project or something you did at work/school. Just a short paragraph is fine, but we’d love to know why you're proud of it.
+2. Tell us about an article/blog post/github repo you found recently, why you liked it, and why we should check it out.
+3. Tell us about a feature of Auth0 you really like, and why.
+
+We also ask them to put together a small [webtask](https://webtask.io/) hack, something that should take a couple of hours.
+
+At this point very few people have thought that investing time to answer this email is not worth it and let us know. More often, we tend to get really interesting responses and learn many things such as:
```
Comment (diegopoza) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: Although there are some people that think that investing time to answer this email is not worthy, more often we tend to get really interesting responses. This responses are useful to learn things about the candidate, such as the following.

**Comment by mgonto** [comment]
Images to CDN you can upload them yourself :D

File: _posts/2016-03-15-how-we-hire-engineers.markdown (hunk 4 above)
Comment (dschenkelman) [inline _posts/2016-03-15-how-we-hire-engineers.markdown]: this should be in past, it refers to the previous step. I think this one is good as is

**Comment by dschenkelman** [comment]
yup, I have permissions. once I have an approved draft I'll change those

**Comment by dschenkelman** [comment]
All done. @diegopoza can I get the spot on Tuesday? Pretty please.

@woloski feel free to push any changes if you would like to.

I'm going to be on vacation, but feel free to merge.

**Pull request merged**